Sources/NIOCore/EventLoopFuture+AssumeIsolated.swift

```diff
@@ -504,22 +587,43 @@ extension EventLoopPromise {
         /// - Parameters:
         ///   - result: The result which will be used to succeed or fail this promise.
         @inlinable
-        func completeWith(_ result: Result<Value, Error>) {
-            self._wrapped.futureResult.eventLoop.assertInEventLoop()
+        @available(*, noasync)
+        public func completeWith(_ result: Result<Value, Error>) {
             self._wrapped._setValue(value: result)._run()
         }
 
         /// Returns the wrapped event loop promise.
         @inlinable
-        func nonisolated() -> EventLoopPromise<Value> {
+        public func nonisolated() -> EventLoopPromise<Value> {
             self._wrapped
         }
     }
 
-    /// Assumes the calling context is isolated to the promise's event loop.
-    @usableFromInline
-    func assumeIsolated() -> Isolated {
+    /// Returns a variant of this ``EventLoopPromise`` with less strict
+    /// `Sendable` requirements. Can only be called from on the
+    /// ``EventLoop`` to which this ``EventLoopPromise`` is bound, will crash
+    /// if that invariant fails to be met.
+    @inlinable
+    @available(*, noasync)
+    public func assumeIsolated() -> Isolated {
+        self.futureResult.eventLoop.preconditionInEventLoop()
+        return Isolated(_wrapped: self)
+    }
+
+    /// Returns a variant of this ``EventLoopPromise`` with less strict
+    /// `Sendable` requirements. Can only be called from on the
+    /// ``EventLoop`` to which this ``EventLoopPromise`` is bound, will crash
+    /// if that invariant fails to be met.
+    ///
+    /// This is an unsafe version of ``EventLoopPromise/assumeIsolated()`` which
+    /// omits the runtime check in release builds. This improves performance, but
+    /// should only be used sparingly.
+    @inlinable
+    public func assumeIsolatedUnsafeUnchecked() -> Isolated {
```

Comment:
```suggestion
    @available(*, noasync)
    public func assumeIsolatedUnsafeUnchecked() -> Isolated {
```

Reply:
Same note here, I want to leave the unsafe unchecked methods without the annotation.